enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -99,259 +116,217 @@ This is often required for industries such as the US public sector, healthcare o
+#### As a cluster administrator, I want to use a local syslog instance only, with no elasticsearch costs.
 
 ```
-# source definitions
-<source>
-  @type tail
-  @label @CONCAT
-  <label @CONCAT>
-    <filter kubernetes.**>
-      @type concat
-      key log
-      partial_key logtag
-      partial_value P
-      separator \'\'
-    </filter>
-    <match kubernetes.**>
-      @type relabel
-      @label @INGRESS
-    </match>
-  </label>
-</source>
- 
-# INGRESS
-<label @INGRESS>
-  ... #current @INGRESS config
-
-    <match output_ops_tag journal.** system.var.log** mux.ops audit.log**   %OCP_FLUENTD_TAGS%>
-      @type relabel
-      @label @_LOGS_INFRA
-    </match>
-    <match kubernetes.**>
-      @type relabel
-      @label @_LOGS_APP
-    </match>
-</label>
-
-<label @_LOGS_APP>
-  <match **>
-     @type copy
-     <store>
-        @type relabel
-        @label @CONTAINER_LOGS
-     </store>
-  </match>
-</label>
-
-<label @_LOGS_INFRA>
-  <match **>
-     @type copy
-     <store>
-        @type relabel
-        @label @INFRA_LOGS
-     </store>
-  </match>
-</label>
-
-# match outputs are generated based on defined pipeline endpoints
-<label @CONTAINER_LOGS>
-  <match **>
-     @type copy
-     #store section for each endpoint type
-     <store>
-       @type relabel
-       @label @SECUREFORWARD_OFFCLUSTER
-     </store>
-     <store>
-       @type relabel
-       @label @ELASTICSEARCH
-     </store>
- </match>
-</label>
-
-<label @CONTAINER_LOGS>
-  <match **>
-     @type copy
-     #store section for each endpoint type
-     <store>
-       @type relabel
-       @label @ELASTICSEARCH_INSECURE
-     </store>
- </match>
-</label>
- 
-#specific pipeline endpoint
-<label @ELASTICSEARCH>
-  <match retry_elasticsearch>
-     @type copy
-     <store>
-       @type elasticsearch
-       @id elasticsearch
-       client_key /var/run/ocp-collector/secrets/elasticsearch/key
-       client_cert /var/run/ocp-collector/secrets/elasticsearch/cert
-       ca_file /var/run/ocp-collector/secrets/elasticsearch/cacert
-       <buffer>
-		@type file
-		path '/var/lib/fluentd/elasticsearch'
-       </buffer>
-     </store>
-  </match>
-  <match **>
-     @type copy
-     <store>
-       @type elasticsearch
-       @id elasticsearch
-       client_key /var/run/ocp-collector/secrets/elasticsearch/key
-       client_cert /var/run/ocp-collector/secrets/elasticsearch/cert
-       ca_file /var/run/ocp-collector/secrets/elasticsearch/cacert
-       retry_tag "retry_elasticsearch"
-     </store>
- </match>
-</label>
-<label @SECUREFORWARD_OFFCLUSTER>
-   ...
-</label>
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+   - name: MyLogs
+     type: Syslog
+     syslog:
+       Facility: Local0
+     url: localstore.example.com:9200
+  pipelines:
+   - inputs: [Infrastructure, Application, Audit]
```

Comment:
IMO it would be good to require a named output as it's easier to ref in the status

Reply:
We do have named outputs and inputs - Infrastructure, Application and Audit are conceptually unique pre-defined input names (with definitions "somewhere else", not visible to user) so you don't have to define them in your CR. For GA we don't have any user-defined inputs, later that's where we'll put selectors.

Reply:
I do wonder if a better api would be:

```
- inputs:
  Infrastructure: true/false
  Application: true/false
  Audit: true/false
```

as it allows for better validation and oc explain will actually show you the names that are valid.

As it stands, it won't be obvious 
a) what the valid values are for output
b) if i've mistyped one of them.


Of course what I am suggesting has drawbacks as well, notably having to keep adding new fields for new inputs.

My proposed api would also reopen the conversation about having a unique CRD for dedicated-admins (it would only have the `pipeline.application` field)

Anyway think about it.

Reply:
I suppose even with the array you can do validation on the array values and maybe with appropriate godoc on the inputs field, it can be made clear what the valid values are..  so i guess i will retract this suggestion.